### PR TITLE
Add hit entity to ProjectileHitEvent

### PR DESCRIPTION
**Pull request by MisterVector**
We know what projectile hit something in ProjectileHitEvent but we don't know what it hit. This adds a hit entity so we can grab what was hit.

### CODE REVIEW

**Review by zachbr — CHANGES_REQUESTED** [review]
Nullable annotations missing

**Comment by AlfieC** [comment]
Unrequired. Projectiles are entities - EntityDamageByEntity event applies here.

**Comment by MiniDigger** [comment]
well, if we have that event anyways, why not add all useful info we can get? I :+1:

File: Spigot-Server-Patches/0173-Add-hit-entity-to-ProjectileHitEvent.patch [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+         return !event.isCancelled();
+     }
+ 
+@@ -782,6 +782,15 @@ public class CraftEventFactory {
+         return event;
+     }
+ 
++    // Paper start
++    public static ProjectileHitEvent callProjectileHitEvent(Entity entity, Entity hitEntity) {
++        ProjectileHitEvent event = new ProjectileHitEvent((Projectile) entity.getBukkitEntity(), (hitEntity == null ? null : (org.bukkit.entity.Entity) hitEntity.getBukkitEntity()));
```
Comment (kashike) [inline Spigot-Server-Patches/0173-Add-hit-entity-to-ProjectileHitEvent.patch:96]: Why do you have a cast to `org.bukkit.entity.Entity` here? It shouldn't be required.

Comment (MisterVector) [reply Spigot-Server-Patches/0173-Add-hit-entity-to-ProjectileHitEvent.patch:96]: If you have a look at CraftEventFactory, there are several org.bukkit.entity.Entity references as the current scope is at the NMS entity, not the bukkit one.

**Comment by zachbr** [comment]
Merged via 96b534f1f1ecfbfb6c8958392a84a36f7e6c0d8f

**Pull request closed without merging**